[RPC] Relax some profiling tests

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49983 [RPC] Relax some profiling tests**

We have observed very rare flakiness in some profiling tests recently,
i.e.: https://github.com/pytorch/pytorch/issues/49635, https://github.com/pytorch/pytorch/issues/49634, https://github.com/pytorch/pytorch/issues/49633. Interestingly, they all seem to have failed on the same job: `pytorch_linux_xenial_cuda10_2_cudnn7_py3_multigpu_test`. However, we were not able to reproduce these even with thousands of
runs on the CI machines where the failure was originally reported. As a result,
relaxing these tests and re-enabling them to reduce failure rates.

Differential Revision: [D25739416](https://our.internmc.facebook.com/intern/diff/D25739416/)

rohan-varma committed:
[RPC] Relax some profiling tests

We have observed very rare flakiness in some profiling tests recently,
i.e.: . However, we were not able to reproduce these even with thousands of
runs on the CI machines where the failure was originally reported. As a result,
relaxing these tests and re-enabling them to reduce failure rates.

Differential Revision: [D25739416](https://our.internmc.facebook.com/intern/diff/D25739416/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] Relax some profiling tests"


We have observed very rare flakiness in some profiling tests recently,
i.e.: https://github.com/pytorch/pytorch/issues/49635, https://github.com/pytorch/pytorch/issues/49634, https://github.com/pytorch/pytorch/issues/49633. Interestingly, they all seem to have failed on the same job: `pytorch_linux_xenial_cuda10_2_cudnn7_py3_multigpu_test`. However, we were not able to reproduce these even with thousands of
runs on the CI machines where the failure was originally reported. As a result,
relaxing these tests and re-enabling them to reduce failure rates.

Differential Revision: [D25739416](https://our.internmc.facebook.com/intern/diff/D25739416/)

[ghstack-poisoned]

mrshenli commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -1579,8 +1583,8 @@ def _profiler_test_with_rpc(self, rpc_exec_mode, func, args, use_record_function
-                self.assertTrue(scope_event.time_range.start < rpc_event.time_range.start)
-                self.assertTrue(scope_event.time_range.end > rpc_event.time_range.end)
+                self.assertLessEqual(scope_event.time_range.start, rpc_event.time_range.start)
+                self.assertGreaterEqual(scope_event.time_range.end, rpc_event.time_range.end)
```
curious, did real tests hit equality here?

rohan-varma replied:
Unfortunately, the stack trace only gives us:

```
Dec 18 23:45:14   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 1583, in _profiler_test_with_rpc
Dec 18 23:45:14     self.assertTrue(scope_event.time_range.end > rpc_event.time_range.end)
Dec 18 23:45:14   File "/opt/conda/lib/python3.6/unittest/case.py", line 682, in assertTrue
Dec 18 23:45:14     raise self.failureException(msg)
Dec 18 23:45:14 AssertionError: False is not true
Dec 18 23:45:14 exiting process with exit code: 10
```

and the error has not reproduced since then. Though I remember seeing some other profiling tests hit equality, possibly due to the actual numerical difference being very small.

mrshenli commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -1799,9 +1809,12 @@ def test_server_process_global_profiler(self):
+        self._assert_top_level_events(inner_events, expected_inner_events)
         outer_events = rpc.rpc_sync(dst_worker_name, get_events_from_profile, (outer_profile_rref,))
-        self._assert_top_level_events(outer_events, ['aten::add', 'aten::sub'])
+        self._assert_top_level_events(outer_events, expected_outer_events)
```
this is doing the same thing as the old code, right?

rohan-varma replied:
Yes, this is just changing variable names, the real change for this test is to just log the discrepancy in events in case of a failure, which will give us more of an idea into what we need to debug if this is ever hit again.

mrshenli reviewed (approved):
LGTM!

rohan-varma committed:
Update on "[RPC] Relax some profiling tests"


We have observed very rare flakiness in some profiling tests recently,
i.e.: https://github.com/pytorch/pytorch/issues/49635, https://github.com/pytorch/pytorch/issues/49634, https://github.com/pytorch/pytorch/issues/49633. Interestingly, they all seem to have failed on the same job: `pytorch_linux_xenial_cuda10_2_cudnn7_py3_multigpu_test`. However, we were not able to reproduce these even with thousands of
runs on the CI machines where the failure was originally reported. As a result,
relaxing these tests and re-enabling them to reduce failure rates.

Differential Revision: [D25739416](https://our.internmc.facebook.com/intern/diff/D25739416/)

[ghstack-poisoned]
